### ISSUE TITLE
Integrate mailing list in application

### ISSUE THREAD

**Issue by vronk** [acdh-oeaw/dhcr-main#51]
- [x] 1. compare the list in the db with the list in the mailserv
      check for users who subscribed directly throug mailman
- [x]  2. Investigate what happened in the past, which problems exist and fix them. 
- [x] 3. Implement Integrate/sync mailing list with application database. 

Edit 2023-05-09 PA: The application already has a subscribe/unsubscribe option (in dashboard, profile settings). When the user changes it there, the change should also be made in the "external" mailing list application (mailman).

**Comment by patrickakk** [comment]
Task 1 & 2 has been discussed and the JFX at 31-1 and can be marked as finished.

**Comment by patrickakk** [comment]
@PixlTracer @IvdL22 @vronk 
Which users should be able to receive the newsletter?

As addition on what we discussed today, I would suggest:
All users, which:
-Their account wasn't disabled
-Are approved (by a moderator or admin)

This way we would have the procedure described here https://dhcr.clarin-dariah.eu/follow#newsletter  also reflected in the application/code. As well we avoid that somebody with a "spam account" receives the newsletter.

> National moderators, lecturers or programme directors who add or curate course metadata are invited to join our contributor mailing list to stay informed about the latest developments. If you do not have an account yet, please register first.

Can we agree on this specifications? Can everybody reply here?

**Comment by IvdL22** [comment]
@patrickakk @vronk @PixlTracer Since the invitation to sign up to the mailing list/newsletter is open only for **those who create an account**, maybe we should hide the option from the public page and make it available only in the moderators and contributors dashboard. 

@PixlTracer For a clean start, I think we should first send an email to **all those whose account is still active** and invite them to subscribe to the DHCR mailing list if they are not yet subscribed. But first we need to make sure that this is working properly. In all future emails, we should have a note at the bottom saying "if you wish to unsubscribe from this mailing list, click here."

As for the **users without an account**, they can sign up only for **the course alert**. @patrickakk Do you know how many users have registered for this one? And is the course alert actually working?

![image](https://user-images.githubusercontent.com/103500506/216003808-5b50222f-b96a-4b2a-8055-1deb82d160a1.png)

**Comment by patrickakk** [comment]
@IvdL22 
Hiding from the public part, what is shown in the picture, sounds like a good idea to me.

**Course reminders**
44 users have subscribed, but only 26 have confirmed their subscription. So the answer is 26.
No, they currently don't work. There is a separate issue for that. Can we continue the talk about that there?
https://github.com/acdh-oeaw/dhcr-main/issues/15

**Comment by patrickakk** [comment]
@PixlTracer  @IvdL22 

REMINDER:

Still open:

> Which users should be able to receive the newsletter?
> 
> As addition on what we discussed today, I would suggest:
> All users, which:
> -Their account wasn't disabled
> -Are approved (by a moderator or admin)
> 
> Can we agree on this specifications?

**Comment by IvdL22** [comment]
@patrickakk @PixlTracer Yes, Patrick, all users with an **active, approved account** should receive the newsletter.

**Comment by IvdL22** [comment]
@patrickakk I think you should change the status of this task in accordance with the next step or close that task only after you have implemented it, up to you.

**Comment by patrickakk** [comment]
@IvdL22 In February you wrote: "maybe we should hide the option from the public page and make it available only in the moderators and contributors dashboard."  (see above).

Then I replied and agreed. Now I think that it's better to keep the explanation the way it is. Would that be ok for you?

**Comment by IvdL22** [comment]
@patrickakk Ok, I agree, thank you.

**Comment by patrickakk** [comment]
Successfully tested in dev by me. Subscribing and unsubscribing works; the changes are immediately available in mailman admin.

The guard will be enabled, so that changes to subscription status can only be made in production. This is required to keep the application db synced with mailman.

Development finished.
Implemented in 2023-05.
This feature can first be tested/reviewed when this release goes into production. (As exception).

Issue blocked until 2023-04 and 2023-05 are released.

**Important**: When 2023-05 is released, all the users who have subscribed in the application, need to be imported in mailman. This is only needed once. Then this feature keeps both synced.

@IvdL22 @PixlTracer See above as info.

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer 
What do we want to do with the registration form here:
https://dev-dhcr.clarin-dariah.eu/users/register

There is the checkbox "Yes, subscribe to the Contributor Mailing List!". Can this be removed? That has two advantages:
- Only users who are approved can subscribe.
- When the sync process with mailman doesn't work, the user registration process won't be affected.

It also has one disadvantage:
- Users have to click extra to subscribe, after their account is approved.

Currently this checkbox on the registration form is not synced with mailman, so we need to take a decision here.

What do you suggest?

**Comment by patrickakk** [comment]
Update: When the user checks that box, this can also be synced with mailman.
In case there is an error in mailman, it can be disabled in the user's profile.

Remaining question: How to deal with the situation where a user subscribes during registration, this is synced with mailman and then the user account shouldn't be approved?

**Comment by IvdL22** [comment]
@patrickakk Thank you, Patrick. Could you set this up in such a way that syncing with mailman occurs only if the user account is approved?

**Comment by IvdL22** [comment]
@patrickakk Regarding the button in the registration form, could we integrate this into the automatic email that confirms that their account has been approved? This way they can sign up if they want to after they register. The button is also available in their dashboard.

**Comment by patrickakk** [comment]
@IvdL22 Yes, that can be done that the sync with mailman is done when the mod/admin approves the account. 
So, we keep the checkbox to subscribe at the registration form?
Checking the box does only enable this in the application database.
When a mod/admin approves the account, it will be synced with mailman. When an error with mailman occurs at this point, the subscription will be disabled in the application database (=profile settings for the user)

Does that answer both of your comments?

Is this the solution we are looking for?

@PixlTracer Do you also agree with that?

**Comment by IvdL22** [comment]
Hi @patrickakk Fine, thank you for the explanation.

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer 

The following process (from above) has been implemented

> When a mod/admin approves the account, it will be synced with mailman. When an error with mailman occurs at this point, the subscription will be disabled in the application database (=profile settings for the user) 

So, updating the status of this issue to:

> Development finished.

> Implemented in 2023-05.
> This feature can first be tested/reviewed when this release goes into production. (As exception).
> 
> Issue blocked until 2023-04 and 2023-05 are released.
> 
> Important: When 2023-05 is released, all the users who have subscribed in the application, need to be imported in mailman. This is only needed once. Then this feature keeps both synced.

**Comment by patrickakk** [comment]
@PixlTracer 
(cc @IvdL22 )

Can you check the settings in the Mailman admin tool?

At general settings:

- [x]  At administrator Tanja is still listed. Can be removed?

- [x] At moderator Iulliana is listed twice (two different mail addresses). This might result in all email notifications being send twice (=more work to process)? Maybe remove one?

**Comment by patrickakk** [comment]
@IvdL22  @PixlTracer  

Info: Today all current subscribed users will be added to Mailman.
The four criteria are:

- User checked box "subcribe to contributor mailing list"
- User account is approved by moderator/admin
- The user has verified their email address. (This prevents fake subscriptions).
- The user account is not disabled by admin.

This also reflects the implementation described above in this issue. Except in case of disabling a user account, we have to manually unsubscribe a user. But this are exceptions.

**Comment by patrickakk** [comment]
@PixlTracer  @IvdL22 

Can you review this?

You can login to the Mailman admin tool and check the members under "Membership Management", "Membership List". This should match your profile settings. (Your email address is there when subscribed, otherwise not).

Then go to: Dashboard / Profile Settings / Sign up to the Mailing List
or directly here: https://dhcr.clarin-dariah.eu/dashboard/profile-settings

Change your subscription status and click on "Update your choice". Now the change is also visible in Mailman.
Note: This only works in production!

Can you change the label to Done if you're satisfied?

**Comment by patrickakk** [comment]
@PixlTracer @IvdL22 

Important note:
Since both of you have access to the Mailman admin tool:
### Please do not change the Mailman admin password now or in the future!

Because that would break the synchronization between the application and Mailman.

**Comment by PixlTracer** [comment]
this is wonderful, seeing the list grow, thank you Patrick!
I guess, we can start then with sending upcoming release notes to the list?

I'll set the labe to done.

**Comment by patrickakk** [comment]
Implemented in 2023-05.

@PixlTracer 
It's up to you what content you send to the subscribers ;)


Can you also answer the two questions above? See:
https://github.com/acdh-oeaw/dhcr-main/issues/51#issuecomment-1610883435

**Comment by PixlTracer** [comment]
done! (ticked the boxes too :) )

**Comment by patrickakk** [comment]
Implemented in 2023-05.
